File: MusicEventManagementSystem/MusicEventManagementSystem.Client/src/frontend/shared/types/api/checkout.ts
import type { PaymentMethod } from '../../../ticket-sales/types/enums/TicketSales';
import type { CartItemDto } from './cart';
import type { OrderTicketDto } from './order';

export interface BillingInfoDto {
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  address?: string;
  city?: string;
  postalCode?: string;
  country?: string;
}

export interface PaymentInfoDto {
  cardNumber: string;
  expiryDate: string;
  cvv: string;
  cardholderName: string;
}

export interface CheckoutRequestDto {
  applicationUserId: string;
  paymentMethod: PaymentMethod;
  promoCode?: string;
  cartItems: CartItemDto[];
  // billingInfo: BillingInfoDto;
  // paymentInfo: PaymentInfoDto;
}

export interface CheckoutResponseDto {
  orderId: number;
  success: boolean;
  message?: string;
  orderNumber?: string;
  totalAmount?: number;
  orderDate?: string;
  status?: string;
  tickets?: OrderTicketDto[];
}

export interface CheckoutFormData {
  paymentMethod: PaymentMethod;
  promoCode?: string;
  // Ovo je samo za prikaz u formi:
  cardholderName?: string;
  cardNumber?: string;
  expiryDate?: string;
  cvv?: string;
}

File: MusicEventManagementSystem/MusicEventManagementSystem.Core/Models/DTOs/Client/Checkout.cs
using MusicEventManagementSystem.Core.Enums.TicketSales;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicEventManagementSystem.Core.Models.DTOs.Client
{
    public class CheckoutRequestDto
    {
        public string ApplicationUserId { get; set; } = string.Empty;
        public PaymentMethod PaymentMethod { get; set; }
        public string? PromoCode { get; set; }
        public List<CartItemDto> CartItems { get; set; } = new();
    }

    public class CheckoutResponseDto
    {
        public int OrderId { get; set; }
        public string OrderNumber { get; set; } = string.Empty;
        public decimal TotalAmount { get; set; }
        public DateTime OrderDate { get; set; }
        public string Status { get; set; } = string.Empty;
        public List<OrderTicketDto> Tickets { get; set; } = new();
    }
}
